### PR TITLE
refactor(login): don't open desktop when login from web

### PR DESCRIPTION
**Pull request by hyrious**

### CODE REVIEW

File: src/v1/controller/login/github/Callback.ts [hunk 1]
```diff
@@ -138,6 +144,10 @@ export const callbackSchemaType: FastifySchema<CallbackRequest> = {
                 type: "string",
                 format: "uuid-v4",
             },
+            platform: {
+                type: "string",
+                nullable: true,
+            },
```
Comment (BlackHole1) [inline src/v1/controller/login/github/Callback.ts:150]:
```suggestion
            needArouse: {
                type: "boolean",
                nullable: true,
            },
```

File: src/v1/controller/login/github/Callback.ts [hunk 2]
```diff
@@ -118,6 +123,7 @@ export const callback: Controller<CallbackRequest, any> = async ({ req, logger }
 
 interface CallbackRequest {
     querystring: {
+        platform?: 'web';
```
Comment (BlackHole1) [inline src/v1/controller/login/github/Callback.ts:126]:
```suggestion
        needArouse?: boolean;
```

**Review by BlackHole1 — CHANGES_REQUESTED** [review]

File: src/v1/controller/login/github/Callback.ts [hunk 3]
```diff
@@ -16,7 +16,7 @@ export const callback: Controller<CallbackRequest, any> = async ({ req, logger }
         "content-type": "text/html",
     });
 
-    const { state: authUUID } = req.query;
+    const { state: authUUID, platform } = req.query;
```
Comment (BlackHole1) [inline src/v1/controller/login/github/Callback.ts:19]:
```suggestion
    const { state: authUUID, needArouse } = req.query;
```

File: src/v1/controller/login/github/Callback.ts [hunk 4]
```diff
@@ -108,7 +108,12 @@ export const callback: Controller<CallbackRequest, any> = async ({ req, logger }
             60 * 60,
         );
 
-        return reply.send(successHTML);
+        let replyHTML = successHTML
+        if (platform === "web") {
```
Comment (BlackHole1) [inline src/v1/controller/login/github/Callback.ts:112]:
```suggestion
        if (needArouse) {
```

File: src/v1/controller/login/github/Callback.ts [hunk 5]
```diff
@@ -108,7 +108,12 @@ export const callback: Controller<CallbackRequest, any> = async ({ req, logger }
...
+            replyHTML = replyHTML.replace(/^\s*setTimeout.*$/, '')
```
Comment (BlackHole1) [inline src/v1/controller/login/github/Callback.ts:113]:
```suggestion
            // remove the recall code
            replyHTML = replyHTML.replace(/^\s*setTimeout.*$/, '')
```

**Review by BlackHole1 — CHANGES_REQUESTED** [review]

File: src/v1/controller/login/github/Callback.ts (hunk 1 above)
Comment (crimx) [reply src/v1/controller/login/github/Callback.ts:150]: I think both choice are ok. But if we need to pick one I think `platform` is more decoupled with business logic comparing to `needArouse` (or `needLaunchApp`) which is an imperative action.

Comment (BlackHole1) [reply src/v1/controller/login/github/Callback.ts:150]: emmmm ok

**Review by BlackHole1 — APPROVED** [review]

File: src/v1/controller/login/github/Callback.ts (hunk 5 above)
Comment (BlackHole1) [reply src/v1/controller/login/github/Callback.ts:113]: add comment

**Review by BlackHole1 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**